Plugins/NativeCamera/Source/NativeCamera.cpp

```diff
@@ -0,0 +1,117 @@
+#include <napi/napi.h>
+#include "NativeCamera.h"
+#include "NativeVideo.h"
+
+#include <Babylon/JsRuntime.h>
+#include <GraphicsImpl.h>
+#include <vector>
+#include <algorithm>
+
+namespace Babylon::Plugins::Internal
+{
+    class NativeCamera : public Napi::ObjectWrap<NativeCamera>
+    {
+        static constexpr auto JS_NAVIGATOR_NAME = "navigator";
+        static constexpr auto JS_CLASS_NAME = "_NativeCamera";
+        static constexpr auto JS_NATIVECAMERA_CONSTRUCTOR_NAME = "NativeCamera";
+
+    public:
+
+        static void Initialize(Napi::Env env)
+        {
+            Napi::HandleScope scope{ env };
+
+            Napi::Function func = NativeCamera::DefineClass(
+                env,
+                JS_CLASS_NAME,
+                {
+                    NativeCamera::InstanceMethod("createVideo", &NativeCamera::CreateVideo),
+                    NativeCamera::InstanceMethod("updateVideoTexture", &NativeCamera::UpdateVideoTexture),
+                });
+
+            JsRuntime::NativeObject::GetFromJavaScript(env).Set(JS_NATIVECAMERA_CONSTRUCTOR_NAME, func);
+
+            // create or get global navigator object
+            Napi::Object global = env.Global();
+            Napi::Object navigator;
+            if (global.Has(JS_NAVIGATOR_NAME))
+            {
+                navigator = global.Get(JS_NAVIGATOR_NAME).As<Napi::Object>();
+            }
+            else
+            {
+                navigator = Napi::Object::New(env);
+                global.Set(JS_NAVIGATOR_NAME, navigator);
+            }
+
+            // append media devices to navigator
+            Napi::Object mediaDevices = Napi::Object::New(env);
+            mediaDevices.Set("getUserMedia", Napi::Function::New(env, [](const Napi::CallbackInfo& info) {
+                auto env = info.Env();
+                auto deferred{ Napi::Promise::Deferred::New(env) };
+                auto promise{ deferred.Promise() };
+               
+                auto& jsRuntime{ JsRuntime::GetFromJavaScript(env) };
+                    jsRuntime.Dispatch([deferred{ std::move(deferred) }](Napi::Env env) {
+                        deferred.Resolve(env.Null());
+                    });
```

Comment:
Why do we need this dispatch? We're already on the JS thread, so this is just waiting for the next JS frame to resolve the promise. Do we need to do that?